src/content/insertable-streams/webgpu/js/multi_video_worker.js

```diff
@@ -0,0 +1,248 @@
+            { texture: videoTexture, origin: { x: 0, y: 270 } },
+            {
+                // the width of the image being copied
+                width: videoBitmap.width,
+                height: videoBitmap.height,
+            }
+        );
+        videoBitmap.close();
+        videoFrame.close();
+    }
+    if (gumFrame) {
+        gumBitmap = await createImageBitmap(gumFrame, { resizeWidth: 480, resizeHeight: 270 });
+        device.queue.copyExternalImageToTexture(
+            { source: gumBitmap, origin: { x: 0, y: 0 } },
+            { texture: videoTexture, origin: { x: 480, y: 0 } },
+            {
+                width: gumBitmap.width,
+                height: gumBitmap.height,
+            }
+        );
+        gumBitmap.close();
+        gumFrame.close();
+    }
+    const uniformBindGroup = device.createBindGroup({
+        layout: renderPipeline.getBindGroupLayout(0),
+        entries: [
+            {
+                binding: 0,
+                resource: sampler,
+            },
+            {
+                binding: 1,
+                resource: videoTexture.createView(),
+            },
+        ],
+    });
+
+    const commandEncoder = device.createCommandEncoder();
+    const textureView = context.getCurrentTexture().createView();
+    const renderPassDescriptor = {
+        colorAttachments: [
+            {
+                view: textureView,
+                loadValue: { r: 0.0, g: 0.0, b: 0.0, a: 1.0 },
+                storeOp: 'store',
+            },
+        ],
+    };
+
+    const passEncoder = commandEncoder.beginRenderPass(renderPassDescriptor);
+    passEncoder.setPipeline(renderPipeline);
+    passEncoder.setVertexBuffer(0, vertexBuffer);
+    passEncoder.setBindGroup(0, uniformBindGroup);
+    passEncoder.draw(6, 1, 0, 0);
+    passEncoder.endPass();
+    device.queue.submit([commandEncoder.finish()]);
+}
+
+onmessage = async (event) => {
+    const { operation } = event.data;
+    if (operation === 'init') {
+        const { canvas } = event.data;
+        context = canvas.getContext('webgpu');
+        if (context === null || context === undefined) {
+            const errorMessage = 'Your browser does not support the WebGPU API.' +
+                ' Please see the note at the bottom of the page.';
+            postMessage({ error: errorMessage });
+            return;
+        }
+        const adapter = await navigator.gpu.requestAdapter();
+        device = await adapter.requestDevice();
+        if (device === null || device === undefined) {
+            console.log('[WebGPUWorker] requestDevice failed.')
+            return;
+        }
+        const swapChainFormat = 'bgra8unorm';
+
+        const rectVerts = new Float32Array([
+            1.0, 1.0, 0.0, 1.0, 0.0,
+            -1.0, -1.0, 0.0, 0.0, 1.0,
+            -1.0, 1.0, 0.0, 0.0, 0.0,
+            1.0, 1.0, 0.0, 1.0, 0.0,
+            1.0, -1.0, 0.0, 1.0, 1.0,
+            -1.0, -1.0, 0.0, 0.0, 1.0,
+        ]);
```

Comment:
I don't really understand why the coordinates are different between the main thread and the worker. Is there an extra flip happening somewhere? Seems like it might be a Chrome bug.

Other than these coordinates, it seems like this file and multi_video_main.js are very similar. Did you consider passing in a canvas and an error callback to WebGPUTransform so that it's usable on both main thread and worker rather than copy/pasting?

(Just in case the problem was with loading multiple files on the worker, check out https://developer.mozilla.org/en-US/docs/Web/API/WorkerGlobalScope/importScripts.)